出題がシャッフルされていない不具合を修正する

YumaInaura opened the pull request:
- 表題の不具合を修正した。
- 1人のユーザーが3連続で出題されてしまう不具合を修正した。
- クイズのテストをリファクタリングした。
# English

slack の #english 窓でアドバイスをいただいた。

![2015-10-02 14 46 32](https://cloud.githubusercontent.com/assets/13635059/10240247/74bba282-6914-11e5-9319-bbfb353e8ffc.png)

Commit:
fix not shuffled questions

Commit:
fix bug, 3 questions per 1 user

- 1人のユーザーあたり、3回連続で出題されてしまう不具合を修正した
  - 画像の登録シチュエーション数分だけ出題されてしまっていた。テストでは画像は1種類しか登録しないため、再現していなかった。
- I fixed bug , 3 questions was asked per 1 user.
  - Question was asked per situation kinds of images num. But in test, user has only 1 situation image only. So I could not find it.

yamotonalds commented on spec/features/quizees_spec.rb:
```diff
@@ -40,7 +39,7 @@
 
     feature 'クイズに回答する' do
-      before do
+       before do
```
インデント

Yinaura replied:
:pray:

yamotonalds commented on spec/features/quizees_spec.rb:
```diff
@@ -50,13 +49,32 @@
+        end
+
+        scenario 'questions are shuffled' do
+            expect(question_user_ids).not_to eq question_user_ids.sort
```
インデント

Yinaura replied:
:pray:

yamotonalds commented on spec/features/quizees_spec.rb:
```diff
@@ -50,13 +49,32 @@
         expect(page).to have_content question_user.detail
       end
 
+      context 'continuly' do
```
continulyって英語として正しいんでしょうか？

Yinaura replied:
continuity でした

hattori0407 replied:
contextは「何がどういう状態の時」みたいな説明になるはずで、
そう書けない場合はsenarioかfeatureをネストするかになるのではと思いました
（何のコンテキストなのかよくわからなかった）

hattori0407 replied:
あとfeature specはもう日本語統一でいいのではと思いました

Yinaura replied:
scenario が適切でした m_ _ m

Yinaura replied:
「もう」のニュアンスはどんな感じでしょうか :question:

Yinaura replied:
context は不適切でした m_ _ m

hattori0407 replied:
- 他が全部日本語で書いてあるので無理に英語で書く必要はない
- feature specは混みいった説明を書くことが多いので、英語で読み書きするコストが高め

みたいな感じでした

Yinaura replied:
わかりにくかったので to many questions continuity という表現にしました :+1:

yamotonalds replied:
too ?

Yinaura replied:
>feature specは混みいった説明を書くことが多いので、英語で読み書きするコストが高め

feature だけ日本語で書くのは、そういう意味だったんですね。

Yinaura replied:
(answer ) to many questions continuity 

という文脈だったのですが、この書き方は分かりにくいでしょうか。

yamotonalds replied:
あ、すみません。全然文脈読んでませんでした。 :bow:

Yinaura replied:
日本語と英語入り混じってたので、typo かどうかも分かりにくい状態でした :pray:

Yinaura replied:
>他が全部日本語で書いてあるので無理に英語で書く必要はない
>feature specは混みいった説明を書くことが多いので、英語で読み書きするコストが高め

ありがとうございます。

英語で spec を書くと、文脈を考えやすく、結果的に適切な context でテストが出来るんじゃないかと感じました。

英語日本語が混在しているのは良くないので、新しいコミットでは英語に揃えてみました :+1:

Commit:
brush up feature test

- Changed language, Write in Japanese to English. Make result of 'rspec -fd' appropriate.
  featureのテストを英語に変えて、ネストを整えた。英語のほうが正しい文脈を考えやすい。 rspec -fd でなるべく適切な文脈がとれるようにした。

---------------------------------------
Result of `rspec -fd`
---------------------------------------

Quiz
  Start
    display starting form
    When the user clicks start button
      the quiz starts
  Answer
    To many questions continuity
      all questions have each different users
      all questions are shuffled
    To one question
      display answer form
      User challenges to a game and get result
        display be answered user profile
        When the answer was incorrect
          display losing message
          display total incorrect num
        When the answer was correct
          display total correct num
          display winning message

hattori0407 commented:
![+1](https://cloud.githubusercontent.com/assets/7765989/10240243/6d534590-6914-11e5-800a-8073c3149b4c.gif)
:+1:

Yinaura commented:
LGTM? ありがとうございます :smile: 

テストでの .push の書き間違えも直しました :pray:

Commit:
delete needless codes

- 不要なコードを削除した

yamotonalds commented:
[![LGTM](http://lgtm.in/p/YQTZu4jgq)](http://lgtm.in/i/YQTZu4jgq)
